### PR TITLE
SEME-1983-BE-Rubocop-petal-with-all-rubocop-config-for-new-projects

### PR DESCRIPTION
**Pull request by nelalampetal**
## What's the PR about

[SEME-1983-BE-Rubocop-petal-with-all-rubocop-config-for-new-projects](https://petalmd.atlassian.net/browse/SEME-1983)

## Work done

Unified cops used by monolith to be reused by seme-microservice and new projects.

## How to test
-The new project should use this gem `rubocop-petal` and inherit the base `config/default.yml` configuration, declaring the heritage in the `.rubocop.yml` as show below:

```
require:
- rubocop-rails
- rubocop-performance
- rubocop-rspec
- rubocop-petal
inherit_mode:
  merge:
  - Exclude
inherit_gem:
  rubocop-petal: config/default.yml

```

If necessary implement any specific bypass or rule, should be declared in the following section like in this example below in the `.rubocop.yml` file from the project that uses the `rubocop-petal` gem

```
# ********************************
# ToDo items
# ********************************
RSpec/EmptyExampleGroup:
  Exclude:
    - 'spec/requests/v1/authentication_api_spec.rb'
    - 'spec/requests/v1/conversations_api_spec.rb'
```

ps: for now is not possible to apply this new version of `rubocop-petal` gem in the `petalmd.rails` project, due many differences of versions from dependencies.

### CODE REVIEW

File: lib/rubocop/petal/version.rb [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 module RuboCop
   module Petal
-    VERSION = '0.9.0'
+    VERSION = '1.0.0'
```
Comment (Bhacaz) [inline lib/rubocop/petal/version.rb:5]: I'll will handle the version once merge. just add an entry in the changelog under `master` and remove change relative to bumping the gem version.

**Review by Bhacaz — CHANGES_REQUESTED** [review]

**Review by Bhacaz — CHANGES_REQUESTED** [review]
You should add as dependency 

* rubocop-performance
* rubocop-rspec

in the gemspec, so the cop will come when simply using `rubocop-petal`

File: lib/rubocop/petal/inject.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ def self.defaults!
         path = CONFIG_DEFAULT.to_s
         hash = ConfigLoader.send(:load_yaml_configuration, path)
         config = Config.new(hash, path).tap(&:make_excludes_absolute)
-        puts "configuration from #{path}" if ConfigLoader.debug?
+        Rails.logger.info "configuration from #{path}" if ConfigLoader.debug?
```
Comment (Bhacaz) [inline lib/rubocop/petal/inject.rb:14]: We are not in a Rails context here. I think `puts` is the good way, Plus, this is auto generated file by rubocop generator.

**Review by Bhacaz — CHANGES_REQUESTED** [review]

File: lib/rubocop/petal/version.rb (hunk 1 above)
Comment (nelalampetal) [reply lib/rubocop/petal/version.rb]: okay

**Comment by Bhacaz** [comment]
An example I followed: https://github.com/github/rubocop-github#usage

I updated the readme to show how import the config in an other project.

**Review by Bhacaz — APPROVED** [review]

File: config/base.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  Enabled: true
+Layout/MultilineHashKeyLineBreaks:
+  Enabled: true
+Layout/MultilineMethodArgumentLineBreaks:
+  Enabled: true
+Lint/AmbiguousBlockAssociation:
+  Exclude:
+    - spec/**/*
+Metrics/AbcSize:
+  Max: 20
```
Comment (IMillerPetal) [inline config/base.yml:33]: @Bhacaz  Did you revisit all the cops rules?
This gem is meant to be a standard for all projects, monolith can override some of them to handle tech debt, right?

Comment (Bhacaz) [reply config/base.yml:33]: Yes I check it all. Monolith have 40. 20 seams a good compromise

Comment (IMillerPetal) [reply config/base.yml:33]: Great. Thanks!

**Comment by nelalampetal** [comment]
@Bhacaz I've tried to include this new version 0.9.0 in our project, but I think we are missing any process to keep this 
 `/config/base.yml` file when bundle the gem. 
Please see the output dir list just after install the gem

```
Installing rubocop-petal 0.9.0
Bundle complete! 18 Gemfile dependencies, 103 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
nelalam@petal-0232 secure-messaging % ls -lrt /Users/nelalam/.rbenv/versions/3.1.4/lib/ruby/gems/3.1.0/gems/rubocop-petal-0.9.0/config/*
-rw-r--r--@ 1 nelalam  staff  2412 May 15 14:48 /Users/nelalam/.rbenv/versions/3.1.4/lib/ruby/gems/3.1.0/gems/rubocop-petal-0.9.0/config/default.yml
```
The result is 

```
nelalam@petal-0232 secure-messaging % rubocop 
Configuration file not found: /Users/nelalam/.rbenv/versions/3.1.4/lib/ruby/gems/3.1.0/gems/rubocop-petal-0.9.0/config/base.yml
```

**Pull request merged**